### ISSUE TITLE
new Date('YYYY-M-D') returns an Invalid Date object only when debugger is off

### ISSUE THREAD

**Issue by thomasttvo** [facebook/react-native#27891]
`new Date('1996-1-1')` returns Invalid Date while the debugger is off. 
If you turn on remote JS debugging, it works fine.
If you pass in `1991-01-01` then it works fine no matter if the JS debugging is on or off

React Native version: "0.60.6"

**Comment by akshetpandey** [comment]
Its the difference between V8 and JSC. You can repro it in chrome vs safari.
This is a know difference between Date parsing implementations of different JS engines. Either use moment.js or use `new Date (year, month [, date [, hours [, minutes [, seconds [, ms ] ] ] ] ] )` constructor. More info here: https://ecma-international.org/ecma-262/5.1/#sec-15.9.3.1

TL;DR date parsing sucks in js, use moment.js

**Comment by hramos** [comment]
As @akshetpandey said, it's due to different JS environments.
